From 6436e8b3ce482759a1d755a3f72a06cfa35263a5 Mon Sep 17 00:00:00 2001
From: Ruslan Bay <ruslanbey@proton.me>
Date: Thu, 26 Feb 2026 17:21:32 +0100
Subject: [PATCH 43/60] media: intel/ipu4: Add guards for uninitialized array
 cleanup

Add NULL checks before cleaning up arrays that may not have been allocated.
This prevents potential crashes when the isys_unregister_subdevices() cleanup
path runs on devices where initialization failed or was incomplete.
---
 drivers/media/pci/intel/ipu-isys.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-isys.c b/drivers/media/pci/intel/ipu-isys.c
index 75085e5d4..d4c1ea5d6 100644
--- a/drivers/media/pci/intel/ipu-isys.c
+++ b/drivers/media/pci/intel/ipu-isys.c
@@ -215,11 +215,13 @@ static void isys_unregister_subdevices(struct ipu_isys *isys)
 
 	ipu_isys_isa_cleanup(&isys->isa);
 
-	for (i = 0; i < tpg->ntpgs; i++)
-		ipu_isys_tpg_cleanup(&isys->tpg[i]);
+	if (isys->tpg)
+		for (i = 0; i < tpg->ntpgs; i++)
+			ipu_isys_tpg_cleanup(&isys->tpg[i]);
 
-	for (i = 0; i < csi2->nports; i++)
-		ipu_isys_csi2_cleanup(&isys->csi2[i]);
+	if (isys->csi2)
+		for (i = 0; i < csi2->nports; i++)
+			ipu_isys_csi2_cleanup(&isys->csi2[i]);
 }
 
 static int isys_register_subdevices(struct ipu_isys *isys)
-- 
2.55.0

